src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -85,6 +85,8 @@ protected void reinitialize() {
    * @param loaderId Id to used for the loader
    */
   protected <D> void startLoad(DataSource<D> dataSource, int loaderId, DataConsumer<D> consumer) {
+    // Kill all previously loaded data by the activity
+    getActivity().getSupportLoaderManager().destroyLoader(loaderId);
```

Comment:
Please add a TODO here. We may need to do something smarter here by using a different kind of loader. Destroying the cached data is not something we want. What we should aim for is invalidating the data if the underlying data changes.

What is the use case in which this returns the stale data?